### PR TITLE
MINOR: Add Jenkinsfile for CK branch builder

### PR DESCRIPTION
**Pull request by jarekr**
Branch builder should show up at https://jenkins.confluent.io/job/confluentinc/job/kafka
Jenkinsfile is copied from ce-kafka. There doesn't appear to be a better way to test other than checking this in, unfortunately.

### CODE REVIEW

**Comment by ijuma** [comment]
Weird error:

> java.lang.UnsatisfiedLinkError: org.rocksdb.BloomFilter.createNewBloomFilter(IZ)J
> 	at org.rocksdb.BloomFilter.createNewBloomFilter(Native Method)
> 	at org.rocksdb.BloomFilter.<init>(BloomFilter.java:74)
> 	at org.rocksdb.BloomFilter.<init>(BloomFilter.java:34)
> 	at org.apache.kafka.streams.state.internals.RocksDBStoreTest.setUp(RocksDBStoreTest.java:82)

**Comment by ijuma** [comment]
@confluentinc/streams ^

**Comment by ableegoldman** [comment]
Ack, looking into it

**Comment by cmccabe** [comment]
LGTM

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    // Per KAFKA-7524, Scala 2.12 is the default, yet we currently support the previous minor version.
+    stage("Check compilation compatibility with Scala 2.11") {
+        sh "gradle"
+        sh "./gradlew clean compileJava compileScala compileTestJava compileTestScala " +
+                "--stacktrace -PscalaVersion=2.11"
+    }
+
+    stage("Compile and validate") {
+        sh "./gradlew clean assemble spotlessScalaCheck checkstyleMain checkstyleTest spotbugsMain " +
+                "--stacktrace --continue -PxmlSpotBugsReport=true"
```
Comment (cmccabe) [inline Jenkinsfile]: need `-x test` to avoid running tests inadvertently

**Comment by ijuma** [comment]
Thanks @ableegoldman!

**Pull request merged**